lldb/source/Plugins/ObjectFile/Minidump/ObjectFileMinidump.cpp

```diff
@@ -65,56 +66,52 @@ bool ObjectFileMinidump::SaveCore(const lldb::ProcessSP &process_sp,
   if (!process_sp)
     return false;
 
-  MinidumpFileBuilder builder;
-
-  Target &target = process_sp->GetTarget();
-
-  Log *log = GetLog(LLDBLog::Object);
-  error = builder.AddSystemInfo(target.GetArchitecture().GetTriple());
-  if (error.Fail()) {
-    LLDB_LOG(log, "AddSystemInfo failed: %s", error.AsCString());
+  llvm::Expected<lldb::FileUP> maybe_core_file = FileSystem::Instance().Open(
```

Comment:
Let's add a block of comment explaining the high level of the serialization logic. Most importantly:
1. Physical layout (header first, directories next, memory list has to be the last and why)
2. Later fixup needs to be done to header/directories after everything to the important fields
3. Later fixup needs to be done to thread.stack field after dumping memory list so that we can avoid duplication